lib/routes/snapshots.js

```diff
@@ -38,8 +39,12 @@ router.get('/:snapshotId', function (req, res) {
 });
 
 router.get('/', function (req, res) {
-  req.log.info('Request to get all snapshots');
-  snapshots.getSnapshots({}, function onGetSnapshots (err, snapshots) {
+  var host = { host: req.query.host || undefined };
+  var url = { requestedUrl: req.query.url || undefined };
+  var o = _.merge(host, url);
```

Comment:
I think the merge is an extra step - you could have:
var opts = _.omitBy({
  host: req.query.host || undefined
  url: req.query.url || undefined
}, _.isNil);